src/DGtal/topology/VoxelComplex.h

```diff
@@ -0,0 +1,621 @@
+  * cannot check concepts here.
+  *
+  */
+
+template < typename TKSpace, typename TObject, typename TCellContainer =
+               typename TKSpace::template CellMap< VoxelComplexCellData >::Type >
+class VoxelComplex :
+  public CubicalComplex< TKSpace, TCellContainer >
+{
+public:
+  // The TObject::DigitalSet::Container must be associative.
+  BOOST_CONCEPT_ASSERT(( concepts::CSTLAssociativeContainer< typename TObject::DigitalSet::Container > ));
+
+  using Self            = VoxelComplex< TKSpace, TObject, TCellContainer > ; ///< Type of this instance of VoxelComplex.
+
+    friend Self& DGtal::functions::operator-=<>( Self&, const Self& );
+    friend Self  DGtal::functions::operator- <>( const Self&, const Self& );
+    // ----------------------- associated types ------------------------------
+  using Parent          = CubicalComplex< TKSpace, TCellContainer > ; ///< Type of the parent class CubicalComplex.
+  using KSpace          = TKSpace ;  ///< Type of the cellular grid space.
+  using CellContainer   = TCellContainer ; ///< Type for storing cells, an associative container Cell -> Data
+  using Data            = typename CellContainer::mapped_type ; ///< Type of data associated to each cell.
+  using Object          = TObject ; ///< Type for input Object storing the digital set of spels with a topology.
+  using DigitalTopology = typename TObject::DigitalTopology ; ///< Type for the type of the digital topology.
+
+  /// The dimension of the embedding space.
+  static const Dimension dimension = KSpace::dimension;
+  using Integer              = typename KSpace::Integer ; ///< Type for integers in the space.
+  using Cell                 = typename KSpace::Cell ;    ///< Type for a cell in the space.
+  using Cells                = typename KSpace::Cells ;   ///< Type for a sequence of cells in the space.
+  using Space                = typename KSpace::Space ;   ///< Type of the digital space
+  using Size                 = typename KSpace::Size ;    ///< Type for a number of elements
+  using Point                = typename KSpace::Point ;   ///< Type for a point in the digital space
+  using DirIterator          = typename KSpace::DirIterator ; ///< Type for iterating over cell directions
+  using CellMap              = CellContainer ; ///< Type for storing cells, an associative container Cell -> Data
+  using CellMapConstIterator = typename CellMap::const_iterator ; ///< Const iterator for visiting type CellMap
+  using CellMapIterator      = typename CellMap::iterator ; ///< Iterator for visiting type CellMap
+
+  //Clique alias
+  using Clique = Parent;
+  using CliqueContainer = std::vector<Clique>;
+
+  //Tables
+  using ConfigMap = boost::dynamic_bitset<>;
+  using PointToMaskMap = std::unordered_map<Point, unsigned int>;
+public:
+  /// Inherit all constructors from parent CubicalComplex.
+  using CubicalComplex<KSpace, CellContainer>::CubicalComplex;
+
+    // #<{(|*
```

Comment:
cleanup the copy constructor ? Or mark it as delete ?

Reply:
Added the copy constructor.

Reply:
Make default constructor protected (as in parent class: CubicalComplex)